sphinx_gallery/gen_rst.py

```diff
@@ -1126,9 +1127,17 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf, seen_backrefs=No
     target_file = os.path.join(target_dir, fname)
     _replace_md5(src_file, target_file, "copy", mode="t")
 
-    file_conf, script_blocks, node = split_code_and_text_blocks(
+    if fname.endswith(".py"):
+        parser = py_source_parser
+        language = "Python"
+    else:
+        parser = BlockParser(fname, gallery_conf)
+        language = parser.language
```

Comment:
On the one hand this is nice because it means whatever `py_source_parser` code we have so far will continue to be used for Python flies, so we're unlikely to break stuff for people.

On the other hand it seems like this will lead to code dup and it might be better to build whatever Python-specific stuff we need (if any? maybe the script_blocks part or something?) into `BlockParser` directly. That could be a follow-up PR I guess because maybe it's not so trivial and/or has other disadvantages (and would definitely be easier to review).

Reply:
Yes, there is definitely some code duplication caused by trying to leave `py_source_parser` as it is. I'd be happy to refactor it as a subclass of `BlockParser` in a follow-up PR, or as part of this one, whichever you prefer.

Reply:
Let's do it as a follow-up PR. If we decide later it was a bad idea it makes reverting easier, but hopefully we don't end up there!